tests/test_write_json.py

```diff
@@ -0,0 +1,650 @@
+"""test_write_json.py"""
+__author__ = "Marius Pahl"
+
+import json
+import tempfile
+import unittest
+from collections import OrderedDict
+
+import numpy as np
+import pandas as pd
+
+from collect_stata.write_json import (
+    generate_stat,
+    stat_dict,
+    stats_cat,
+    stats_number,
+    stats_string,
+    uni,
+    uni_cat,
+    uni_statistics,
+    write_json,
+)
+
+
+class TestWriteJson(unittest.TestCase):
+    """
+    Tests for all methods from write_json.py
+    """
+
+    def setUp(self):
+        """
+        Set a TemporaryDirectory for the json testfile
+        """
+        self.sandbox = tempfile.TemporaryDirectory()
+        return super().setUp()
+
+    @staticmethod
+    def get_testdatatable():
+        """
+        Create a test dataframe
+        """
+        data_table = pd.DataFrame()
+        data_table["TESTCAT"] = [-1, -1, 1, 2, 1, np.nan, 1, 2, 1, 2, 2, 1, 1, 2, 2]
+        data_table["TESTSTRING"] = [
+            "",
+            "a",
+            "b",
+            ".",
+            np.nan,
+            "c",
+            np.nan,
+            ".",
+            np.nan,
+            "d",
+            "e",
+            "f",
+            "f",
+            "f",
+            "g",
+        ]
+        data_table["TESTNUMBER"] = [-1, -1, -2, 5, 10, 10, 15, 100, 10, 2, -1, 3, 4, 5, 6]
+        data_table["TESTOTHER"] = [
+            -1,
+            "a",
+            -2,
+            5,
+            "b",
+            np.nan,
+            15,
+            "x",
+            2,
+            3,
+            "b",
+            1,
+            2,
+            "y",
+            "z",
+        ]
+
+        return data_table
+
+    @staticmethod
+    def get_testmetadata():
```

Comment:
Please don't create your static test data in your code like this.
Store it in a json file or in another file format that supports the structure of your data and then load it from there.

Reply:
done